test/transaction.js

```diff
@@ -78,41 +162,48 @@ it('NULL_DATA() should be deterministic', function () {
+
   // var multiSigAddress = zencashjs.address.multiSigRSToAddress(redeemScript, '2092')
 
-  const blockHeight = 15600
-  const blockHash = '0214c87f3f06ab6a22325da40f3c2066838fd50e75e1e0dc0205935fcbb79ec8'
+  const blockHeight = 0
+  const blockHash = '0da5ee723b7923feb580518541c6f098206330dbc711a6678922c11f2ccf1abb'
 
   var txobj = zencashjs.transaction.createRawTx(
     [{
-      txid: 'd6f04de4f1ab745d8d7d3d6846b718b3bef4baf857af4f26e3847162317982d9',
+      txid: '4e8ab22e49dff3ab23890d1569d704a07b0b5dba1bb3bbeb827f8a4a738abd55',
       vout: 0,
       scriptPubKey: ''
     }],
-    [{address: 'ztimpo6bUJk8ngMpRXf3yyhTZBMDLQrDQJD', satoshis: 5000000000}
+    [{address: 'ztUVGgccByusDMSrxNsUtA4pr7nUNUHJjgK', satoshis: 1143740000}
     ],
     blockHeight,
     blockHash
   )
 
   // Prepare our signatures for mutli-sig
-  var sig1 = zencashjs.transaction.multiSign(txobj, 0, privKeys[0], redeemScript)
-  var sig2 = zencashjs.transaction.multiSign(txobj, 0, privKeys[1], redeemScript)
-  var tx0 = zencashjs.transaction.applyMultiSignatures(txobj, 0, [sig1, sig2], redeemScript)
+  var sig1 = zencashjs.transaction.multiSign(txobj, 0, privKeys[2], redeemScript)
+  var sig2 = zencashjs.transaction.multiSign(txobj, 0, privKeys[3], redeemScript)
+
+  expect(sig1).to.equal('3044022051f3d6e86fab4c3f2ab6f26b3df8611c7d4202feeb22ac11f20646847bae2fb102202653d332897cf065e8ba600f2ba953945c26e05d1131c0d7934a3d1f6659e88901')
+  expect(sig2).to.equal('30450221008ab128c0556ee941df0663917a523c2ec3aecc102190ee5ccff776e97d11163b02205b1c3ea546b5e4b1895c85ccbb84c7ef381af0d723a6aedddb314fd24800723e01')
 
+  var tx0 = zencashjs.transaction.applyMultiSignatures(txobj, 0, [sig1, sig2], redeemScript)
   // Serialize the transaction
-  var serializedTx = zencashjs.transaction.serializeTx(tx0)
+  var serializedTx = zencashjs.transaction.serializeTx(tx0, true)
+  expect(serializedTx).to.equal('010000000155bd8a734a8a7f82ebbbb31bba5d0b7ba004d769150d8923abf3df492eb28a4e0000000000fd410100473044022051f3d6e86fab4c3f2ab6f26b3df8611c7d4202feeb22ac11f20646847bae2fb102202653d332897cf065e8ba600f2ba953945c26e05d1131c0d7934a3d1f6659e889014830450221008ab128c0556ee941df0663917a523c2ec3aecc102190ee5ccff776e97d11163b02205b1c3ea546b5e4b1895c85ccbb84c7ef381af0d723a6aedddb314fd24800723e014cad522102c397ea717172d4d3eabfe5989a9b6f6107eeac7d949766ea97770a34db2dd9912103fd5f6e6a732bef2b24ec753de0c44a4d7228b34cac4d55aed19b196dcfcdeb7221025c65e267a0848ed8247267aecff6eab5f4955b28fe25d3ca37874b710788c65721020493561cea1b2fb6e358e9e3cce20bb1c01ab56c599078cc02b461b452bd5a212103c4da5ce479219d563b40037069e094e698644f5ba947bd9f20420c1d27268baa55aeffffffff0160162c44000000003c76a9140e9efda56509425ab71914a4deae722db91fc92888ac20bb1acf2c1fc1228967a611c7db30632098f0c641855180b5fe23793b72eea50d00b400000000')
```

Comment:
Please spend real transaction inputs in the tests and broadcast the transactions. That way they can be independently verified using zen-cli getrawtransaction/decoderawtransaction.

Testnet can be used so no real zen have to be spent.

Reply:
I'm not sure about serialization compatibility between zencashjs and zen core.
I mean, that serializeTx(...) will give us different result than getrawtransaction(...)